tests/src/core/testqgslayoutmap.cpp

```diff
@@ -458,6 +459,39 @@ void TestQgsLayoutMap::dataDefinedStyles()
   QVERIFY( checker.testLayout( mReport, 0, 0 ) );
 }
 
+void TestQgsLayoutMap::dataDefinedCrs()
+{
+  QList<QgsMapLayer *> layers = QList<QgsMapLayer *>() << mRasterLayer << mPolysLayer << mPointsLayer << mLinesLayer;
+
+  QgsLayout l( QgsProject::instance() );
+  l.initializeDefaults();
+
+  QgsLayoutItemMap *map = new QgsLayoutItemMap( &l );
+  map->attemptMove( QgsLayoutPoint( 20, 20 ) );
+  map->attemptResize( QgsLayoutSize( 200, 100 ) );
+  map->setFrameEnabled( true );
+  map->setLayers( layers );
+  l.addLayoutItem( map );
+
+  QString def_crs = map->extent().asWktCoordinates();
+
+  //test epsg variable
+  map->dataDefinedProperties().setProperty( QgsLayoutObject::MapCrs, QgsProperty::fromExpression( QStringLiteral( "EPSG:2192" ) ) );
```

Comment:
```suggestion
  map->dataDefinedProperties().setProperty( QgsLayoutObject::MapCrs, QgsProperty::fromStaticValue( QStringLiteral( "EPSG:2192" ) ) );
```

`EPSG:2192` isn't a valid QGIS expression :wink:  (same elsewhere in this test)